### DISCUSSION TITLE
Where can I set the MainPage in an async method at application start?

### DISCUSSION THREAD

**Discussion by asusralis** [dotnet/maui#2593] [Q&A]
I need to use an async method to determine whether I should start the application with my shell or a login page. In Forms I used OnStart, but in maui it says the MainPage must be set before this is called.

**Comment by wallym** [comment]
I don't know your scenario.  I did lookup and can see an OnStart listed in the intellisense in VS2022.  Is that of help?

**Reply by asusralis** [reply]
It seems in Maui you must set the MainPage **before** OnStart is called.

**Reply by wallym** [reply]
That's what I figured you were running into.  I assumed it was more complicated than I was thinking.  :-)
